File: Chapter05/5.4 Server Monitor/Before/ServerMonitor.cs
using System;

namespace ServerMonitor.Before
{
    // THE CORE (Tightly Coupled)
    public class ServerMonitor
    {
        public void CheckTemperature(int temp)
        {
            // VIOLATION: Hardcoded check inside the logic
            if (temp > 95)
            {
                // VIOLATION: We are hardcoding a dependency on a specific external tool (Twilio).
                // We can't test this logic without actually sending a text message!
                var twilio = new TwilioClient("API_KEY");
                twilio.SendSms("555-1234", "Server is overheating!");
            }
            else
            {
                Console.WriteLine($"Temp {temp} is nominal.");
            }
        }
    }

    // FAKE 3RD PARTY LIBRARY (To make code compile)
    public class TwilioClient
    {
        public TwilioClient(string key) { }
        public void SendSms(string to, string body) 
        {
            Console.WriteLine($"[Twilio API] Sending SMS to {to}: {body}");
        }
    }

    // PROGRAM ENTRY
    class Program
    {
        static void Main(string[] args)
        {
            var monitor = new ServerMonitor();
            monitor.CheckTemperature(95); // This will actually try to call the API
        }
    }
}

File: Chapter 05/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter05/section_5_4_server_monitor/before/ServerMonitor.java
package com.grokkingsoftwarearchitecture.chapter05.section_5_4_server_monitor.before;

import com.grokkingsoftwarearchitecture.chapter05.shared.LogManager;
/**
 * The Core Business Logic.
 * This class currently fails as a "Boundary Keeper" because it allows
 * external infrastructure details to leak into the domain.
 */
public class ServerMonitor {
    
    /**
     * Evaluates server health based on temperature.
     * @param temp The temperature value to check.
     */
    public void checkTemperature(int temp) {
        // VIOLATION: Hardcoded magic number.
        if (temp > 95) {
            // VIOLATION: Direct Dependency.
            // By hardcoding the 'TwilioClient', we have abandoned our post
            // as a Clarity Engineer.
            TwilioClient twilio = new TwilioClient("API_KEY");
            twilio.sendSms("555-1234", "Server is overheating!");
        } else {
            LogManager.info(ServerMonitor.class, "Temp {0} is nominal.", temp);
        }
    }
}

/**
 * Simulates a third-party Library.
 */
class TwilioClient {
    public TwilioClient(String key) { }
    public void sendSms(String to, String body) {
        LogManager.info(TwilioClient.class, "[Twilio API] Sending SMS to {0}: {1}", to, body);
    }
}